xonsh/environ.py

```diff
@@ -816,17 +819,19 @@ def _yield_executables(directory, name):
 
 
 def locate_binary(name):
-    if os.path.isfile(name) and name != os.path.basename(name):
-        return name
-    directories = builtins.__xonsh_env__.get('PATH')
-    # Windows users expect to be able to execute files in the same directory
-    # without `./`
-    if ON_WINDOWS:
-        directories = [_get_cwd()] + list(directories)
-    try:
-        return next(chain.from_iterable(_yield_executables(directory, name) for
-                    directory in directories if os.path.isdir(directory)))
-    except StopIteration:
+    """Locates an executable on the file system."""
+    if os.path.isfile(name):
+        if ON_WINDOWS:
```

Comment:
I think there is an issue here. Imagine we have a `strombidae.exe` in the current directory. Then the if should trigger no matter what if the user writes `strombidae` or `STROMBIDAE` or  `Strombidae.exe`. Basically, it should also be able to launch any executable with an extension in `PATHEXT`. This thing was previously handled by `executables_in` https://github.com/xonsh/xonsh/blob/master/xonsh/tools.py#L487